File: StudyAsist-Android/app/src/main/java/com/studyasist/data/local/entity/QuestionType.kt
package com.studyasist.data.local.entity

import androidx.room.TypeConverter

enum class QuestionType {
    MCQ,
    FILL_BLANK,
    SHORT,
    ESSAY,
    NUMERIC,
    TRUE_FALSE,
    MATCHING,
    DIAGRAM
}

class QuestionTypeConverter {
    @TypeConverter
    fun from(value: QuestionType): String = value.name

    @TypeConverter
    fun to(value: String): QuestionType = QuestionType.valueOf(value)
}

File: StudyAsist-iOS/StudyAsist/Models/QuestionType.swift
//
//  QuestionType.swift
//  StudyAsist
//

import Foundation

enum QuestionType: String, Codable, CaseIterable {
    case mcq = "MCQ"
    case fillBlank = "FILL_BLANK"
    case short = "SHORT"
    case essay = "ESSAY"
    case numeric = "NUMERIC"
    case trueFalse = "TRUE_FALSE"
    case matching = "MATCHING"
    case diagram = "DIAGRAM"
}
